### PR TITLE
Enable nightly CodeQL scans

### PR DESCRIPTION
**Pull request by samsymons**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/0/1205371338744155/f
iOS PR: N/A
macOS PR: N/A
What kind of version bump will this require?: None, this is a workflow change

CC: @ayoy 

**Description**:

This PR enables nightly CodeQL scans. The iOS equivalent was done here: https://github.com/duckduckgo/iOS/pull/1958

**Steps to test this PR**:
1. Check that this PR has a CodeQL alert visible
1. Remove the code that triggers this alert (see commit history for that)
1. Check that the alert goes away

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16
* [ ] macOS 10.15
* [ ] macOS 11
* [ ] macOS 12

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "sam/codeql" ]
```
Comment (samsymons) [inline .github/workflows/codeql.yml:16]: Temporary change to force code scanning to run on this branch, will remove this after PR approval

File: Sources/BrowserServicesKit/SecureVault/AutofillSecureVault.swift [hunk 2]
```diff
@@ -127,6 +127,14 @@
...
         self.providers = providers
         self.expiringPassword = ExpiringValue(expiresAfter: 60 * 60 * 24 * 3)
+
+
+        let tainted = try! String(contentsOf: URL(string: "http://example.com")!)
+
+        let a = String("abc") // GOOD: not a format string
+        let b = String(tainted) // GOOD: not a format string
+        let c = String(format: "abc") // GOOD: not tainted
+        let d = String(format: tainted) // BAD
```
[the remark this answers is not in the document]
Comment (samsymons) [reply Sources/BrowserServicesKit/SecureVault/AutofillSecureVault.swift:137]: This is an example alert using some vulnerable code from the CodeQL test suite, to demonstrate that alerting is working.

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (NelsonSolis90) [reply .github/workflows/codeql.yml:16]: Good

**Review by ayoy — APPROVED** [review]
LGTM! Thanks @samsymons!

**Pull request merged**